test/test_nestedtensor.py

```diff
@@ -7824,131 +7832,558 @@ def __torch_dispatch__(self, func, types, args=..., kwargs=None):
         self.assertEqual(res.shape, (4, nt.shape[1], 6))
 
 
-FORWARD_FAILURES = {
-    # === BEGIN NotImplementedError SECTION ===
-    # unary
-    "nn.functional.celu",
-    "nn.functional.elu",
-    "nn.functional.hardshrink",
-    "nn.functional.hardsigmoid",
-    "nn.functional.hardtanh",
-    "nn.functional.logsigmoid",
-    "nn.functional.mish",
-    "nn.functional.relu6",
-    "nn.functional.rrelu",
-    "nn.functional.selu",
-    "nn.functional.softplus",
-    "nn.functional.softshrink",
-    "nn.functional.threshold",
-    "rad2deg",
-    # binary
-    "__rsub__",
-    "complex",
-    "floor_divide",
-    "polar",
-    "rsub",
-    # reduction
-    "count_nonzero",
-    "linalg.vector_norm",
-    "nansum",
-    "std",
-    "std.unbiased",
-    "var",
-    "var.unbiased",
-    # === BEGIN UNSUPPORTED SECTION ===
-    # ValueError: expects strided tensor (got torch.jagged tensor)
-    "masked.amax",
-    "masked.amin",
-    "masked.argmax",
-    "masked.argmin",
-    "masked.logsumexp",
-    "masked.mean",
-    "masked.norm",
-    "masked.prod",
-    "masked.std",
-    "masked.sum",
-    "masked.var",
-    # === BEGIN BUG SECTION ===
-    # Returns a tuple of Tensors so it doesn't work with NJT's unary pointwise logic
-    "frexp",
+# The following lists specify rules indicating how to handle particular SampleInputs: are they
+# expected to fail, should they be skipped, etc. Note that rules are attempted to be matched
+# from top to bottom and only one rule at most will be matched, so order matters! The guiding
+# general principle here should be one xfail / skip per bug if at all possible :)
+FORWARD_FAILURES = [
+    # not implemented
+    XFailRule(
+        error_type=NotImplementedError,
+        match_fn=lambda device, dtype, op, sample: op.full_name
+        in {
+            # unary
+            "nn.functional.celu",
+            "nn.functional.elu",
+            "nn.functional.hardshrink",
+            "nn.functional.hardsigmoid",
+            "nn.functional.hardtanh",
+            "nn.functional.logsigmoid",
+            "nn.functional.mish",
+            "nn.functional.relu6",
+            "nn.functional.rrelu",
+            "nn.functional.selu",
+            "nn.functional.softplus",
+            "nn.functional.softshrink",
+            "nn.functional.threshold",
+            "rad2deg",
```

Comment:
huh, Why is rad2deg in this list, but not deg2rad? o-o

Reply:
good question! Looks like it's because `deg2rad` is tagged with `tags: pointwise` and `rad2deg` is not for some reason.. this should probably be fixed.